### ISSUE TITLE
Nvidia driver compatibility

### ISSUE THREAD

**Issue by amarkpayne** [trustin/linux510-qca6390#1]
Hello,

Thanks for making this repo! I can confirm that this works for the XPS17 9700 as well, and the WIFI is stable. Unfortunately, I am having problems getting the nvidia drivers for the RTX 2060 working. I have tried a few solutions online, but so far no luck. My questions are:

1. Is there any modifications to this repo that would cause problems with the nvidia drivers? (Besides the general issues with the switch to the 5.10 kernel)

2. How difficult would it be to try and build these changes off of the 5.9 kernel? If it is possible any advice on how I should proceed?
Thanks!

**Comment by trustin** [comment]
Hi! My XPS13 9310 uses integrated graphs only, so I didn't have a chance to check if it's stable with nVidia drivers. Did you have a chance to install the `nvidia-dkms` package? It should work with any kernel versions since it compiles modules for you.

It seems like `nvidia-dkms` has been removed from AUR recently, so you'll have to use the PKGBUILD at https://pastebin.com/zwZYdRFw and build it by yourself.

Please let me know how it goes. I'm also curious about it. :smile:

**Comment by trustin** [comment]
Alternatively, you could try one of the following packages from AUR:

- nvidia-vulkan-dkms
- nvidia-beta-dkms
- nvidia-dkms-performance

**Comment by amarkpayne** [comment]
Still trying to figure things out. I think I am running into issues because the kernel is named linux510-qca6390, so a lot of the time I get dependency errors. Is there a way to change the name of this kernel? Would that be a bad idea?

**Comment by trustin** [comment]
I've just pushed a commit that adds `linux510` and `linux510-headers` to `provides`. I'll also try to install one of the `nvidia-dkms` packages to see if it works on my PC with RTX 3070. It doesn't have QCA6390, but it should allow me to verify if this kernel works with an nVidia driver.

**Comment by trustin** [comment]
> Is there a way to change the name of this kernel? Would that be a bad idea?

You can do that by modifying PKGBUILD file but I think it's better fixing the dependency issue properly.

**Comment by trustin** [comment]
I was able to install `linux510`, `linux510-qca6390` and `nvidia-beta-dkms` after removing prebuilt nvidia binary packages without any conflicts. Could you confirm it works for you as well, @amarkpayne

**Comment by amarkpayne** [comment]
Sorry for the late reply. So I was able to install the above items, and before rebooting `nvidia-smi` appeared to work. Once I rebooted though I could not make it past the Dell/Manjaro black screen (it froze after a few seconds). Uninstalling nvidia-beta-dkms fixed the issue, but nvidia-smi no longer works because of it

**Comment by amarkpayne** [comment]
I finally figured it out! Documenting it all here in case someone has to figure this out after me.

Part of it seems to have been enabling the nvidia-persistenced daemon with `sudo systemctl enable nvidia-persistenced` (there was some debugging to be done here to get it all working, but I forget the exact steps. Looking through journalctl was helpful if it didn't start). This fixed the issue with `nvidia-smi` hanging after printing. See https://stackoverflow.com/questions/41489070/nvidia-smi-process-hangs-and-cant-be-killed-with-sigkill-either for more details.

This did not fix the issue of the manjaro/Dell load screen freezing upon restart. For this I followed the instructions [here](https://wiki.archlinux.org/index.php/Dell_XPS_17_(9700)) about the Prime render offloader. Specifically I created the file /etc/X11/xorg.conf with the following contents:

```
Section "Device"
  Identifier "iGPU"
  Driver "modesetting"
EndSection

Section "Screen"
  Identifier "iGPU"
  Device "iGPU"
EndSection

Section "Device"
  Identifier "dGPU"
  Driver "nvidia"
EndSection

```

That did the trick and I was able to login and the nvidia drivers were working. I also found installing `prime-run` to be helpful for certain applications.
